src/app/pages/catalog/catalog.component.ts

```diff
@@ -14,9 +15,25 @@ import { AsyncPipe } from '@angular/common';
 })
 export class CatalogComponent implements OnInit {
   private readonly categoriesService = inject(CategoriesService);
+  private readonly quizCategoriesStoreService = inject(
+    QuizCategoriesStoreService,
+  );
+
   categories$!: Observable<QuizCategory[]>;
 
   ngOnInit(): void {
-    this.categories$ = this.categoriesService.getRandomCategories();
+    this.categories$ = this.quizCategoriesStoreService.getCategories();
+
+    this.categories$.subscribe((categories) => {
+      if (!categories || categories.length === 0) {
+        this.loadCategories();
```

Comment:
Components should not know about store manipulation methods. They should only query the data.
Store addition and update should be processed in the API component. You can add there an RxJs tap() operator and update the store from there